R2API/ItemAPI.cs

```diff
@@ -1,514 +1,502 @@
+            ItemDef.canRemove = canRemove;
+            ItemDef.descriptionToken = descriptionToken;
+            ItemDef.hidden = hidden;
+            ItemDef.loreToken = loreToken;
+            ItemDef.name = name;
+            ItemDef.nameToken = nameToken;
+            ItemDef.pickupIconSprite = pickupIconSprite;
+            ItemDef.pickupModelPrefab = pickupModelPrefab;
+            ItemDef.pickupToken = pickupToken;
+            ItemDef.tags = tags;
+            ItemDef.tier = tier;
+            ItemDef.unlockableDef = unlockableDef;
+
+            ItemDisplayRules = new ItemDisplayRuleDict(itemDisplayRules);
+        }
+
+        public CustomItem(string name, string nameToken,
+            string descriptionToken, string loreToken,
+            string pickupToken,
+            Sprite pickupIconSprite, GameObject pickupModelPrefab,
+            ItemTier tier, ItemTag[] tags,
+            bool canRemove,
+            bool hidden,
+            UnlockableDef unlockableDef = null,
+            ItemDisplayRuleDict? itemDisplayRules = null) {
+            ItemDef = ScriptableObject.CreateInstance<ItemDef>();
+            ItemDef.canRemove = canRemove;
+            ItemDef.descriptionToken = descriptionToken;
+            ItemDef.hidden = hidden;
+            ItemDef.loreToken = loreToken;
+            ItemDef.name = name;
+            ItemDef.nameToken = nameToken;
+            ItemDef.pickupIconSprite = pickupIconSprite;
+            ItemDef.pickupModelPrefab = pickupModelPrefab;
+            ItemDef.pickupToken = pickupToken;
+            ItemDef.tags = tags;
+            ItemDef.tier = tier;
+            ItemDef.unlockableDef = unlockableDef;
+
+            ItemDisplayRules = itemDisplayRules;
+        }
+    }
+
+    public class CustomEquipment {
+        public EquipmentDef? EquipmentDef;
+        public ItemDisplayRuleDict? ItemDisplayRules;
+
+        public CustomEquipment(EquipmentDef? equipmentDef, ItemDisplayRule[]? itemDisplayRules) {
+            EquipmentDef = equipmentDef;
+            ItemDisplayRules = new ItemDisplayRuleDict(itemDisplayRules);
+        }
+
+        public CustomEquipment(EquipmentDef? equipmentDef, ItemDisplayRuleDict? itemDisplayRules) {
+            EquipmentDef = equipmentDef;
+            ItemDisplayRules = itemDisplayRules;
+        }
+
+        public CustomEquipment(string name, string nameToken,
+            string descriptionToken, string loreToken,
+            string pickupToken,
+            Sprite pickupIconSprite, GameObject pickupModelPrefab,
+            float cooldown,
+            bool canDrop,
+            bool enigmaCompatible,
+            bool isBoss, bool isLunar,
+            BuffDef passiveBuffDef,
+            UnlockableDef unlockableDef,
+            ColorCatalog.ColorIndex colorIndex = ColorCatalog.ColorIndex.Equipment,
+            bool appearsInMultiPlayer = true, bool appearsInSinglePlayer = true,
+            ItemDisplayRule[]? itemDisplayRules = null) {
```

Comment:
Is there anyway we can do indentation so that someone can see at a glance where the constructor param ends and the constructor code begins?

Reply:
I could do it on my end, but any time we run a formatter it would return to being not being indented since the project settings have no newline between parameters and bracket

Reply:
Can you start the method with a newline?